solutionbox/code_free_ml/mltoolbox/code_free_ml/analyze.py

```diff
@@ -158,6 +158,76 @@ def parse_arguments(argv):
   return args
 
 
+def make_analysis_plan(schema, features):
```

Comment:
Instead of introducing a concept "plan", I wonder if it is clearer if we structure the code such as:

1. Make a collections.DefaultDict(list), key being source column, value is the list of transforms.
2. Check for conflicting transforms in one place (check_schema_transforms_match(), or rename it).
3. During analysis, you only need to get the first transform in each source column since we know all transforms are compatible.

Thoughts?